developer_manual/app_development/dependency_management.rst

```diff
@@ -34,6 +34,58 @@ Dependency hell
 
 Be careful with which packages you add to an app. PHP can not load two version of the same class twice, hence there can be conflicts between Nextcloud Server and an app or between two or more apps if they require the same package. So try to keep the number of production dependencies to a minimum and see :ref:`app-composer-bin-tools`.
 
+Alternatively you can use [composer-bin-plugin](https://github.com/bamarni/composer-bin-plugin) to avoid dependency conflicts between apps.
+
+
+1. Install composer-bin-plugin according to their docs.
+
+```
+	composer require --dev bamarni/composer-bin-plugin
+	composer bin psalm require --dev psalm/phar
+	composer bin psalm require --dev nextcloud/ocp:dev-master
```

Comment:
Split it into require of bin-plugin and installation of psalm, or other needed deps